### PR TITLE
Split personalities can now communicate directly

### PR DESCRIPTION
**Pull request by MCterra10**
## About The Pull Request
This PR allows people with the split personality trauma to communicate with the stranger and vice versa through an action button. I also considered adding an option for the active player to release control but I felt that would be breaking the spirit of brain traumas a bit much.

## Why It's Good For The Game
Split personality has always been a situation where the second player's only real option is "How many times can I shout "I'm Gay"/WGW over radio and try and avoid surgery until I inevitably get removed?" Don't get me wrong, you can still do that, but I aim in this PR to give players another option to get some of that sweet RP in. The main argument I can see coming up against this change is that traumas are supposed to be negatives, but how I see it, this doesn't make Split Personality a positive trauma so much as it allows roleplay similar to how you can roleplay as a paraplegic war veteran or as an ex-security officer who developed a phobia following an 'incident'. I'm happy to add a cooldown to the action if it proves to be an issue.

## Changelog
:cl:
add: Split personalities can now communicate with an action button
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
Make it able to communicate via phones aswell.

**Comment by XDTM** [comment]
I initially opted for no communication to make the personalities more distinct and unable to easily coordinate, even if they share the same goal: to talk to each other they have to wait their turn, even if the current controller is making a mistake and the backseat personality wants to warn them.

The scenario i wanted to mostly avoid is split personality becoming a knockoff imaginary friend, with a pal giving you advice or helping you in situation requiring robustness, instead of being a severe trauma.

However i can also imagine fun situations arising from full communication, especially with split personalities that openly antagonize the current host, so it might be a better option in the end.

**Comment by Learner10** [comment]
I like this idea. It takes out a frustrating detail, which I'm all for.

**Comment by Ratvar** [comment]
*collapse

**Comment by anconfuzedrock** [comment]
the currently active personality should have to talk out loud because they're fucking crazy

**Review by actioninja — CHANGES_REQUESTED** [review]
I agree on that, give it only to the dormant personality, with active needing to talk out loud.

**Comment by MCterra10** [comment]
Alrighty then, will get that done shortly

File: code/modules/spells/spell_types/personality_commune.dm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	charge_max = 0
+	clothes_req = FALSE
+	range = -1
+	include_user = TRUE
+	action_icon_state = "telepathy"
+	action_background_icon_state = "bg_spell"
+	var/datum/brain_trauma/severe/split_personality/trauma
+	var/flufftext = "You hear an echoing voice in the back of your head..."
+	var/boldnotice = "boldnotice"
+	var/notice = "notice"
```
Comment (actioninja) [inline code/modules/spells/spell_types/personality_commune.dm]: ?

File: code/modules/spells/spell_types/personality_commune.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	if(!msg)
+		charge_counter = charge_max
+		return
+	to_chat(user, "<span class='[boldnotice]'>You concentrate and send thoughts to your other self:</span> <span class='[notice]'>[msg]</span>")
+	to_chat(trauma.owner, "<span class='[boldnotice]'>[flufftext]</span> <span class='[notice]'>[msg]</span>")
+	log_directed_talk(user, trauma.owner, msg, LOG_SAY ,"[name]")
+	for(var/ded in GLOB.dead_mob_list)
+		if(!isobserver(ded))
+			continue
+		var/follow = FOLLOW_LINK(ded, user)
```
Comment (actioninja) [inline code/modules/spells/spell_types/personality_commune.dm]: assignment isn't needed here, just embed this directly in the format.

**Review by actioninja — CHANGES_REQUESTED** [review]

File: code/modules/spells/spell_types/personality_commune.dm (hunk 1 above)
Comment (MCterra10) [reply code/modules/spells/spell_types/personality_commune.dm]: That was copied from telepathy, but I'll change it I guess

File: code/modules/spells/spell_types/personality_commune.dm (hunk 2 above)
Comment (MCterra10) [reply code/modules/spells/spell_types/personality_commune.dm]: Alright

**Pull request merged**